### PR TITLE
set ulimit for server according to redhat spec

### LINKED ISSUE #9872: ulimits are ignored for systemd processes
##### ISSUE TYPE
 * Bug Report

##### COMPONENT NAME

~~~
packaging
~~~

##### CLOUDSTACK VERSION

~~~
N/A
~~~

##### CONFIGURATION

N/A

##### OS / ENVIRONMENT

RHEL7+

##### SUMMARY

We are setting max process limits in `/etc/security/limits.conf` to unlimited for cloud user. But RHEL7+ uses systemd which ignores the limits set in `/etc/security/limits.conf` or `/etc/security/limits.d/*.conf`.

##### STEPS TO REPRODUCE

Run `cat /proc/$(ps aux |grep cloudstack-management|grep -v grep |awk '{print $2}')/limits`
This will print the limit for the cloudstack-management process. The max process limit is not set to unlimited.

##### EXPECTED RESULTS

~~~
The max process limit is set to unlimited.
~~~

##### ACTUAL RESULTS

~~~
The max process limit is not set to unlimited.
~~~

### PR DESCRIPTION
**Pull request by DaanHoogland**
### Description

This PR fixes #9872 

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI
- [ ] test (unit or integration test code)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

#### How did you try to break this feature and the system with this change?

Also see original ticket, set the contents of `/etc/systemd/system/cloudstack-management.service.d/filelimit.conf`
```
# vi /etc/systemd/system/cloudstack-management.service.d/filelimit.conf 
# systemctl daemon-reload
# systemctl restart cloudstack-management
# cat /proc/$(ps aux |grep cloudstack-management|grep -v grep |awk '{print $2}')/limits
```

### CODE REVIEW

**Comment by weizhouapache** [comment]
@DaanHoogland 

- is there some changes for debian ?

- I checked my env
```
# cat /proc/$(ps aux |grep cloudstack-management|grep -v grep |awk '{print $2}')/limits
Limit                     Soft Limit           Hard Limit           Units     
...
Max open files            524288               524288               files     
...
```

after applying the changes, run `systemctl daemon-reload` and restarted mgmt server, 
```
# cat /proc/$(ps aux |grep cloudstack-management|grep -v grep |awk '{print $2}')/limits
Limit                     Soft Limit           Hard Limit           Units     
...
Max open files            500000               500000               files     
...
```

it works. but what should be the best value ?

**Comment by DaanHoogland** [comment]
@weizhouapache I think the best value is installation dependant. If it works on debian-likes, there is nothing to do there. This was reported by a user on centos7 and does not pertain to the default value but to the mechs of changing it that didn't work. We can change the value before merging to contain the default. I hadn't looked at that yet. It was just a guestimate.

**Comment by weizhouapache** [comment]
> @weizhouapache I think the best value is installation dependant. If it works on debian-likes, there is nothing to do there. This was reported by a user on centos7 and does not pertain to the default value but to the mechs of changing it that didn't work. We can change the value before merging to contain the default. I hadn't looked at that yet. It was just a guestimate.

it looks like there is no such file for debian/ubuntu. we can skip it for now (better to have).

currently the value for rhel is 4096  which is too small.
```
$ cat packaging/el8/cloud.limits
cloud hard nofile 4096
cloud soft nofile 4096
```

**Comment by vishesh92** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test keepEnv

**Comment by DaanHoogland** [comment]
ping @vishesh92 , I think this is ready (if you agree)

**Comment by vishesh92** [comment]
We also need to set `LimitNPROC` for the management server to update max number of processes.

File: packaging/centos7/filelimit.conf [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# should go in /etc/systemd/system/cloudstack-management.service.d/
+[Service]
+LimitNOFILE=500000
```
Comment (vishesh92) [inline packaging/centos7/filelimit.conf:20]:
```suggestion
LimitNPROC=infinity
```

File: packaging/centos8/filelimit.conf [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# should go in /etc/systemd/system/cloudstack-management.service.d/
+[Service]
+LimitNOFILE=500000
```
Comment (DaanHoogland) [inline packaging/centos8/filelimit.conf:20]:
```suggestion
LimitNPROC=infinity```

**Comment by vishesh92** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Comment by weizhouapache** [comment]
is setting for `LimitNOFILE`  needed ?
@DaanHoogland @vishesh92

**Comment by DaanHoogland** [comment]
I think none of the sessions are needed @weizhouapache , just the facility for operators to do it. The old method/file did not work with systemd, hence the new one.

**Comment by weizhouapache** [comment]
> I think none of the sessions are needed @weizhouapache , just the facility for operators to do it. The old method/file did not work with systemd, hence the new one.

thanks @DaanHoogland 

the file name `filelimit.conf` is a bit out-of-date. 
never mind, let's keep it

**Comment by DaanHoogland** [comment]
> the file name `filelimit.conf` is a bit out-of-date. never mind, let's keep it

ai, you are right. I'll create a rename PR soon.

**Pull request merged**